### PR TITLE
use safe functions provided by PSL

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -15,6 +15,26 @@ parameters:
...
+		-
+			message: "#^Parameter \\#1 \\$iterable of function Psl\\\\Dict\\\\map expects iterable\\<\\(int\\|string\\), array\\|Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>, array\\<array\\<Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>\\|Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>\\|Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField given\\.$#"
+			count: 1
+			path: src/Test/Controller/AdminControllerWebTestCase.php
+
+		-
+			message: "#^Parameter \\#2 \\$second of function Psl\\\\Type\\\\union expects Psl\\\\Type\\\\TypeInterface\\<array\\<array\\<Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>\\|Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>\\>, Psl\\\\Type\\\\TypeInterface\\<array\\<array\\<Symfony\\\\Component\\\\DomCrawler\\\\Field\\\\FormField\\>\\>\\> given\\.$#"
+			count: 1
+			path: src/Test/Controller/AdminControllerWebTestCase.php
+
```
Comment (azjezz) [inline phpstan-baseline.neon]: PSL doesn't use phpstan, i will have to look into this and see what is the issue exactly.

psalm passes as is.

File: src/Controller/BaseCrudController.php [hunk 2]
```diff
@@ -43,7 +42,7 @@ protected function allowOnlyActions(Actions $actions, string ...$allowedActions)
         ];
 
         return $actions->disable(
-            ...array_diff($allActions, $allowedActions)
+            ...Vec\concat($allActions, $allowedActions)
```
Comment (dragosprotung) [inline src/Controller/BaseCrudController.php]: This should be `Dict\diff`(it should actually be from `Vec` namespace but there is no equivalent).

File: src/Test/Controller/AdminControllerWebTestCase.php [hunk 3]
```diff
@@ -60,24 +54,29 @@ protected function prepareAdminUrl(array $queryParameters): string
     }
 
     /**
-     * @param FormField|array<FormField>|FormField[][] $fields
+     * @param FormField|array<array-key, FormField>|array<array-key, array<array-key, FormField>> $fields
```
Comment (dragosprotung) [inline src/Test/Controller/AdminControllerWebTestCase.php:57]: The original annotation was not actually correct. The value of $fields can be N-deep array.

**Comment by dragosprotung** [comment]
Thank you very much for this PR @azjezz . Excellent work. And also learned more about PSL

**Pull request merged**